jax/_src/scipy/eigh.py

```diff
@@ -21,6 +21,32 @@
 from jax import lax
 
 
+def _fill_diagonal(X, vals):
+  return jax.ops.index_update(X, jnp.diag_indices(X.shape[0]), vals)
```

Comment:
JAX style thing: I'd always prefer `X.at[index].update(y)` over `index_update` these days. They are synonyms but I think the terser syntax is more readable.